api_app/analyzers_manager/observable_analyzers/cycat.py

```diff
@@ -0,0 +1,114 @@
+import logging
+import re
+
+import requests
+
+from api_app.analyzers_manager import classes
+from tests.mock_utils import MockUpResponse, if_mock_connections, patch
+
+logger = logging.getLogger(__name__)
+
+
+class CyCat(classes.ObservableAnalyzer):
+    """
+    This analyzer is a wrapper for cycat api.
+    """
+
+    def update(self) -> bool:
+        pass
+
+    url: str = "https://api.cycat.org"
+
+    def uuid_lookup(self, uuid: str):
+        logger.info(f"performing lookup on uuid: {uuid}")
+        response = requests.get(
+            self.url + "/lookup/" + uuid,
+            headers={"accept": "application/json"},
+        )
+        response.raise_for_status()
+        return response.json()
+
+    def run(self):
+        final_response = {}
+        uuid_pattern = re.compile(
+            r"\b[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\b",
+            re.IGNORECASE,
+        )
+        if uuid_pattern.match(self.observable_name):
+            final_response = self.uuid_lookup(self.observable_name)
+
+        else:
+            response = requests.get(
+                self.url + "/search/" + self.observable_name,
+                headers={"accept": "application/json"},
+            )
+            response.raise_for_status()
+            response = response.json()
+            for uuid in response:
+                final_response[uuid] = self.uuid_lookup(uuid)
+        return final_response
+
+    @classmethod
+    def _monkeypatch(cls):
+        patches = [
+            if_mock_connections(
+                patch(
+                    "requests.get",
+                    return_value=MockUpResponse(
+                        [
+                            "24bfaeba-cb0d-4525-b3dc-507c77ecec41",
```

Comment:
I am not sure this is the expected response. I would have expected data, like the one here: https://cycat.org/services/#fetch-item-by-uuid

UUID are useless as an output so I would have expected to get the data for the UUID. Your analyzer seems right to me so I don't understand why you get this result. can you please try again?

Reply:
yes sure, makes sense.